src/app/public/src/modules/table-of-contents/table-of-contents.component.ts

```diff
@@ -9,4 +10,53 @@ import { StacheNavLink } from '../nav';
 export class StacheTableOfContentsComponent {
   @Input()
   public routes: StacheNavLink[];
+  private viewTop: number;
+  private documentBottom: number;
+
+  constructor(
+    private windowRef: StacheWindowRef,
+    private omnibarService: StacheOmnibarAdapterService
+  ) {
+    this.windowRef.scrollEvent.subscribe(() => {
```

Comment:
Make sure we are unsubscribing to events on destroy or some other mechanism when we are finished.